src/msg/async/ProtocolV2.cc

```diff
@@ -2653,6 +2575,11 @@ CtPtr ProtocolV2::handle_reconnect(char *payload, uint32_t length) {
     connection->target_addr = reconnect.addrs().msgr2_addr();
   }
 
+#warning we need to set connection peer_type when we get tag_hello
```

Comment:
Hmm, I'm confused then, because in this branch i see
```
  if (connection->get_peer_type() == -1) {
    connection->set_peer_type(peer_type);

    ceph_assert(state == ACCEPTING);
    connection->policy = messenger->get_policy(peer_type);
    ldout(cct, 10) << __func__ << " accept of host_type " << (int)peer_type
                   << ", policy.lossy=" << connection->policy.lossy
                   << " policy.server=" << connection->policy.server
                   << " policy.standby=" << connection->policy.standby
                   << " policy.resetcheck=" << connection->policy.resetcheck
                   << dendl;
```
and
```
2019-02-05 20:53:41.477 7f80058a5700  1 --2- v2:172.21.15.36:6805/105513 >> v2:172.21.15.36:6801/103777 conn(0x5637ce9ba800 0x5637ce873180 :-1 s=CONNECTING pgs=0 cs=0 l=0)._banner_exchange_handle_peer_banner peer_type=4 supported=0 required=0
```
but the failure i see clearly shows unknown.0.  see
/a/sage-2019-02-05_20:10:27-rados-wip-sage-testing-2019-02-05-1033-distro-basic-smithi/3553485
osd.4

Reply:
nevermind, it's because we didn't set peer_name yet.  removing the warning!